packages/components/src/styles/components/form/field.scss

```diff
@@ -44,8 +44,9 @@
     }
   }
 
-  .hds-form-field__character-count {
-    margin-bottom: 8px;
+  // Add spacing between the character count and the error message if one exists
+  .hds-form-field__character-count + .hds-form-error {
+    margin-top: 8px;
```

Comment:
There were different amounts of spacing between various child elements so I refactored the problem margin style vs. using `gap` to set spacing.